cpp/tests/io/json_test.cpp

```diff
@@ -1370,6 +1371,66 @@ TEST_F(JsonReaderTest, JsonExperimentalLines)
   CUDF_TEST_EXPECT_TABLES_EQUAL(legacy_reader_table.tbl->view(), table.tbl->view());
 }
 
+TEST_F(JsonReaderTest, JsonLongString)
+{
+  // Unicode
+  // 0000-FFFF     Basic Multilingual Plane
+  // 10000-10FFFF  Supplementary Plane
+  cudf::test::strings_column_wrapper col1{
+    {"\"\\/\b\f\n\r\t",
+     "\"",
+     "\\",
+     "/",
+     "\b",
+     "\f\n",
+     "\r\t",
+     "$€",
+     "ராபிட்ஸ்",
+     "C𝞵𝓓𝒻",
+     "",                                                // null
+     "",                                                // null
+     "கார்த்தி",
+     "CႮ≪ㇳ䍏凹沦王辿龸ꁗ믜스폶ﴠ",  //  0000-FFFF
+     "𐀀𑿪𒐦𓃰𔙆 𖦆𗿿𘳕𚿾[↳] 𜽆𝓚𞤁🄰",                            // 10000-1FFFF
+     "𠘨𡥌𢗉𣇊𤊩𥅽𦉱𧴱𨁲𩁹𪐢𫇭𬬭𭺷𮊦屮",                // 20000-2FFFF
+     "𰾑𱔈𲍉"},                                         // 30000-3FFFF
+    cudf::test::iterators::nulls_at({10, 11})};
+
+  cudf::test::fixed_width_column_wrapper<int16_t> repeat_times{
+    1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 20, 40, 80, 160, 320, 640, 1280};
```

Comment:
Just in case we have some issues when the length is not aligned with warps. Feel free to reject, it does look odd.
```suggestion
    1, 2, 3, 4, 5, 6, 7, 8, 9, 13, 19, 37, 81, 161, 323, 631, 1279};
```